app/infrastructure/persistence/repositories/event_repo.py

```
@@ -79,12 +94,73 @@ async def get_last_events_for_subjects(
+        result = await self.db.execute(
+            select(Event).where(
+                Event.tenant_id == tenant_id,
+                tuple_(Event.subject_id, Event.external_id).in_(
+                    list(subject_external_pairs)
+                ),
+            )
+        )
+        rows = result.scalars().all()
+        return {(e.subject_id, e.external_id): _event_to_result(e) for e in rows}
+
+    async def get_by_tenant_and_seq(
```

Flake8 (7.3.0)

[error] redefinition of unused 'get_by_tenant_and_seq' from line 111

(F811)